### PR TITLE
feat(client): dataset cmd of swcli support output to json

### PR DESCRIPTION
**Pull request by jialeicui**
## Modules
- [ ] UI
- [ ] Controller
- [ ] Agent
- [x] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [x] run code format and lint check
- [ ] add unit test
- [ ] add necessary doc

### CODE REVIEW

**Comment by jialeicui** [comment]
image

**Comment by jialeicui** [comment]
#642

File: client/starwhale/base/view.py [hunk 1]
```diff
@@ -201,3 +191,124 @@ def _print_list(
...
+                if _v.get("id"):
+                    _version = f"[{_v['id']:2}] {_version}"
+
+                result.append(
+                    {
+                        "name": _name,
+                        "version": _version,
+                        "tags": _v.get("tags", []),
+                        "size": _v.get("size", 0),
+                        "runtime": _v.get("runtime", ""),
```
Comment (tianweidut) [inline client/starwhale/base/view.py:223]: For the base view's list_data method, some cases may not include runtime column, such as `Runtime` object.

File: client/starwhale/core/dataset/view.py [hunk 2]
```diff
@@ -96,3 +94,53 @@ def tag(self, tags: t.List[str], remove: bool = False, quiet: bool = False) -> N
...
+            project_uri, fullname, show_removed, page, size
+        )
+        cls.pretty_json(_datasets)
+
+    def info(self, fullname: bool = False) -> None:
+        self.pretty_json(self.get_info_data(self.dataset.info(), fullname))
+
+
+def get_term_view(ctx_obj: t.Dict) -> t.Type[DatasetTermView]:
+    if "output" in ctx_obj and ctx_obj["output"] == "json":
```
Comment (tianweidut) [inline client/starwhale/core/dataset/view.py:144]: `if "output" in ctx_obj and ctx_obj["output"] == "json"` -> `ctx_obj.get('output') == "json"` ?

**Review by tianweidut — APPROVED** [review]
LGTM

**Pull request merged**